src/main/java/net/sf/jabref/gui/journals/ManageJournalAbbreviationsView.java

```diff
@@ -0,0 +1,543 @@
+        setUpTable();
+        setBindings();
+        setButtonStyles();
+        new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                loadingLabel.setVisible(true);
+                progressIndicator.setVisible(true);
+                viewModel.createFileObjects();
+                journalFilesBox.getSelectionModel().selectLast();
+                viewModel.addBuiltInLists();
+                loadingLabel.setVisible(false);
+                progressIndicator.setVisible(false);
+            }
+        }).start();
+    }
+
+    private void setButtonStyles() {
+        journalFilesBox.setPromptText(Localization.lang("No abbreviation files loaded"));
+        Text addJournalFileButtonGraphic = new Text(IconTheme.JabRefIcon.OPEN.getCode());
+        addJournalFileButtonGraphic.getStyleClass().add("icon");
+        addJournalFileButton.setGraphic(addJournalFileButtonGraphic);
+        Text addNewJournalFileButtonGraphic = new Text(IconTheme.JabRefIcon.NEW.getCode());
+        addNewJournalFileButtonGraphic.getStyleClass().add("icon");
+        addNewJournalFileButton.setGraphic(addNewJournalFileButtonGraphic);
+        Text removeJournalAbbreviationsButtonGraphic = new Text(IconTheme.JabRefIcon.CLOSE.getCode());
+        removeJournalAbbreviationsButtonGraphic.getStyleClass().add("icon");
+        removeJournalAbbreviationsButton.setGraphic(removeJournalAbbreviationsButtonGraphic);
+        ButtonBar.setButtonData(progressIndicator, ButtonData.LEFT);
+        ButtonBar.setButtonData(loadingLabel, ButtonData.LEFT);
+        ButtonBar.setButtonUniformSize(progressIndicator, false);
+        ButtonBar.setButtonUniformSize(loadingLabel, false);
+    }
+
+    private void setUpTable() {
+        journalAbbreviationsTable.setOnKeyPressed(event -> {
+            if ((event.getCode() == KeyCode.DELETE) && isEditableAndRemovable.get()) {
+                if ((viewModel.currentAbbreviationProperty().get() != null)
+                        && !viewModel.currentAbbreviationProperty().get().isPseudoAbbreviation()) {
+                    viewModel.deleteAbbreviation();
+                }
+            }
+        });
+        journalTableNameColumn.setCellValueFactory(cellData -> cellData.getValue().nameProperty());
+        journalTableNameColumn.setCellFactory(cell -> new JournalAbbreviationsNameTableEditingCell());
+        journalTableAbbreviationColumn.setCellValueFactory(cellData -> cellData.getValue().abbreviationProperty());
+        journalTableAbbreviationColumn.setCellFactory(cell -> new JournalAbbreviationsAbbreviationTableEditingCell());
+        journalTableEditColumn.setCellValueFactory(cellData -> cellData.getValue().isPseudoAbbreviationProperty());
+        journalTableDeleteColumn.setCellValueFactory(cellData -> cellData.getValue().isPseudoAbbreviationProperty());
+        journalTableEditColumn.setCellFactory(column -> new TableCell<AbbreviationViewModel, Boolean>() {
```

Comment:
I really don't like these complicated cell factories...stupid javafx (not your fault and nothing to do 😺 )